File: src/imzdesk/server/schema/filesystem.py
from enum import Enum

from pydantic import BaseModel


class FileType(str, Enum):
    MSI = 'MSI'
    WSI = 'WSI'


class FilesystemEntry(BaseModel):
    directory: bool
    parent: str
    name: str
    path: str
    size: int | None = None
    type: FileType | None = None

File: app/app/types/filesystem.ts
export const FILE_TYPES = ['MSI', 'WSI'] as const // See `imzdesk.server.utils.filesystem:resolve_filetype`

export type FileType = typeof FILE_TYPES[number]

export interface FilesystemEntry {
  directory: boolean
  parent: string
  name: string
  path: string
  size?: number
  type?: FileType | null
}
